command/agent_test.go

```diff
@@ -489,6 +489,124 @@ listener "tcp" {
 	}
 }
 
+// TestAgent_NoAutoAuthTokenIfNotConfigured tests that API proxy will not use the auto-auth token
+// unless configured to.
+func TestAgent_NoAutoAuthTokenIfNotConfigured(t *testing.T) {
+	logger := logging.NewVaultLogger(hclog.Trace)
+	cluster := vault.NewTestCluster(t, nil, &vault.TestClusterOptions{
+		HandlerFunc: vaulthttp.Handler,
+	})
+	cluster.Start()
+	defer cluster.Cleanup()
+
+	serverClient := cluster.Cores[0].Client
+
+	// Unset the environment variable so that proxy picks up the right test
+	// cluster address
+	defer os.Setenv(api.EnvVaultAddress, os.Getenv(api.EnvVaultAddress))
+	os.Unsetenv(api.EnvVaultAddress)
```

Comment:
You can get the test framework to do this for you too in a single line.

```suggestion
	t.Setenv(api.EnvVaultAddress, "")
```

Reply:
This doesn't work in CI for some reason that I'm forgetting, but I tried this before and it works locally but not in CI

Reply:
Hmm, this approach failed locally too. Not sure why, but in either case, I remember we needed to do it this way for CI